### PR TITLE
Improve error handling in ModelStructure

### PR DESCRIPTION
**Pull request by PeterMeisrimelModelon**
Relevant changes:

1.  Any failure in parsing dependencies will now result in the default dependency (as if attribute was missing); dependency on everything.
2. Error checks on attributes for variables corresponding to ModelStructure elements have been relaxed to warnings and no longer invalidate the ModelStructure or interrupt parsing.

### CODE REVIEW

File: Test/FMI3/fmi3_import_model_structure_test.cpp [hunk 1]
```diff
@@ -244,200 +237,305 @@ TEST_CASE("Valid ModelStructure parsing") {
...
+    REQUIRE(var != nullptr);
 
-    const char* logMsg = "EventIndicator ignored since FMU kind is Co-Simulation or Scheduled Excecution.";
-    REQUIRE(fmi3_testutil_log_contains(tfmu, logMsg));
+    fmi3_import_free_variable_list(varList);
 
     fmi3_testutil_import_free(tfmu);
 }
 
-TEST_CASE("Error check: ModelStructure; ClockedState without clocks attribute") {
```
Comment (PeterMeisrimelModelon) [inline Test/FMI3/fmi3_import_model_structure_test.cpp:336]: This test was redundant; the check is already covered by attribute combinations checks in ModelVariables.

File: src/XML/src/FMI3/fmi3_xml_model_structure.c [hunk 2]
```diff
@@ -300,107 +298,121 @@ static int fmi3_xml_parse_dependencies(fmi3_xml_parser_context_t *context,
...
+/* Wrapper for #fmi3_xml_parse_dependencies, handling failed states */
+static void fmi3_xml_parse_dependencies_error_wrapper(fmi3_xml_parser_context_t* context,
+        fmi3_xml_elm_enu_t elmID, fmi3_xml_dependencies_t* deps) {
+    // number of added dependecies(Kind) entries, these are needed to clean dependencies in case of failure to parse
+    size_t numDepInd = 0;
+    size_t numDepKind = 0;
+    size_t totNumDep = jm_vector_get_size(size_t)(&deps->dependenciesVRs); // original size, before parsing dependencies
+
+    if (fmi3_xml_parse_dependencies(context, elmID, deps, &numDepInd, &numDepKind)) {
+        // Dependecy parsing failed, cleanup of already parsed entries
```
Comment (PeterMeisrimelModelon) [inline src/XML/src/FMI3/fmi3_xml_model_structure.c:396]: OPTIONAL: Could add a more explicit message here that the previous error in parsing the dependencies results in using the default dependencies. Not sure if this is helpful or just extra bloat.

Comment (modelonrobinandersson) [reply src/XML/src/FMI3/fmi3_xml_model_structure.c:396]: I think we are fine leaving it as it is for now.

File: src/XML/src/FMI3/fmi3_xml_model_structure.c [hunk 3]
```diff
@@ -300,107 +298,121 @@ static int fmi3_xml_parse_dependencies(fmi3_xml_parser_context_t *context,
...
+        for (size_t k = 0; k < numDepInd; k++) {
+            jm_vector_remove_item(size_t)(&deps->dependenciesVRs, totNumDep);
+        }
+        for (size_t k = 0; k < numDepKind; k++) {
+            jm_vector_remove_item(char)(&deps->dependenciesKind, totNumDep);
+        }
+
+        // Fallback to default dependency, required for functioning API
```
Comment (modelonrobinandersson) [inline src/XML/src/FMI3/fmi3_xml_model_structure.c:404]: Maybe this comment can be explained a bit more. At least from my POV the first question that came into my mind was "Why is it required for a functioning API?". Consider this optional but I think it's good to have for people less involved with the code.

File: src/XML/src/FMI3/fmi3_xml_model_structure.c [hunk 4]
```diff
@@ -477,103 +489,96 @@ int fmi3_xml_handle_ContinuousStateDerivative(fmi3_xml_parser_context_t *context
...
     if (!data) {
-        fmi3_xml_model_description_t* md = context->modelDescription;
-        fmi3_xml_model_structure_t* ms = md->modelStructure;
-
-        return fmi3_xml_parse_unknown(context, fmi3_xml_elmID_InitialUnknown, &ms->initialUnknowns, ms->initialUnknownDeps);
+        if (fmi3_xml_parse_unknown(context, fmi3_xml_elmID_InitialUnknown, &ms->initialUnknowns, ms->initialUnknownDeps)) {
+            return -1;
+        }
+    } else {
+        // post-processing
```
Comment (modelonrobinandersson) [inline src/XML/src/FMI3/fmi3_xml_model_structure.c:541]: I guess no post-processing is necessary is that correct?

**Review by modelonrobinandersson — APPROVED** [review]
Well done!

File: src/XML/src/FMI3/fmi3_xml_model_structure.c (hunk 3 above)
Comment (PeterMeisrimelModelon) [reply src/XML/src/FMI3/fmi3_xml_model_structure.c:404]: Expanded on the comment, please check if this is more helpful now.

File: src/XML/src/FMI3/fmi3_xml_model_structure.c (hunk 4 above)
Comment (PeterMeisrimelModelon) [reply src/XML/src/FMI3/fmi3_xml_model_structure.c:541]: Correct. Respectively, none are implemented yet. 

Just added it as a placeholder here, for consistency will all the other ModelStructure elements.

**Pull request merged**